data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -214,4 +254,9 @@ private void mergeValueWithValues(final Object value, final List<Object> values)
     private boolean shouldBreakOnMatch(final Map<String, Object> captures) {
         return captures.size() > 0 && grokPrepperConfig.isBreakOnMatch();
     }
+
+    private void runWithTimeout(final Runnable runnable) throws TimeoutException, ExecutionException, InterruptedException {
+        Future<?> task = executorService.submit(runnable);
+        task.get(grokPrepperConfig.getTimeoutMillis(), TimeUnit.MILLISECONDS);
```

Comment:
We should update the READMe with this configuration options now that we are using it.